cpp/include/cuvs/preprocessing/spectral/spectral_embedding.hpp

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <cuvs/preprocessing/spectral/spectral_embedding_types.hpp>
+
+#include <raft/core/device_mdspan.hpp>
+#include <raft/core/resources.hpp>
+
+namespace cuvs::preprocessing::spectral {
+
+// template <typename IndexTypeT, typename ValueTypeT>
+auto spectral_embedding(raft::resources const& handle,
+                        raft::device_matrix_view<float, int, raft::row_major> nums,
```

Comment:
We always put params in the order of:

1. Resources
2. Params struct
3. Input mdspan
4. Output mdspan
5. Free params (and defaults) 

Please make this consistent.